src/server/readpool/mod.rs

```diff
@@ -25,49 +24,64 @@ use util;
 use util::futurepool::{self, FuturePool};
 
 pub use self::config::Config;
-pub use self::context::Context;
 pub use self::priority::Priority;
 
 const TICK_INTERVAL_SEC: u64 = 1;
 
-#[derive(Clone)]
-pub struct ReadPool {
-    pool_high: FuturePool<Context>,
-    pool_normal: FuturePool<Context>,
-    pool_low: FuturePool<Context>,
+pub struct ReadPool<T: futurepool::Context + 'static> {
+    pool_high: FuturePool<T>,
+    pool_normal: FuturePool<T>,
+    pool_low: FuturePool<T>,
     max_tasks_high: usize,
     max_tasks_normal: usize,
     max_tasks_low: usize,
 }
 
-impl util::AssertSend for ReadPool {}
-impl util::AssertSync for ReadPool {}
+impl<T: futurepool::Context + 'static> util::AssertSend for ReadPool<T> {}
+impl<T: futurepool::Context + 'static> util::AssertSync for ReadPool<T> {}
 
-impl ReadPool {
-    pub fn new(config: &Config) -> Self {
+impl<T: futurepool::Context + 'static> Clone for ReadPool<T> {
+    fn clone(&self) -> Self {
```

Comment:
why not using `#[derive(Clone)]` direclty?

Reply:
there is a bug that derive Clone will not work as intended: https://github.com/rust-lang/rust/issues/26925